### PR TITLE
fix: omit NSG rules created by LoadBalancer during upgrade

### LINKED ISSUE #1624: Application not accessible from outside cluster during aks-engine Upgrade
Applications are not accessible from outside the cluster during the aks-engine upgrade

**Steps To Reproduce**
1. Deploy application(say nginx) and service of type LoadBalancer to expose the application to be accessed from outside the cluster
2. Upgrade cluster using aks-engine upgrade. During upgrade operation the application will not be accessible from outside the cluster

**Expected behavior**
Should be able to access the application from outside the cluster during upgrade process

**AKS Engine version**
v0.36.2

**Kubernetes version**
1.12.8

**Additional context**

### PR DESCRIPTION
**Pull request by marosset**
…ncer during upgrade

**Reason for Change**:

**Issue Fixed**:

Fixes #1624

**Requirements**:

- [ ] uses [conventional commit messages](https://www.conventionalcommits.org/)
  
- [ ] includes documentation
- [ ] adds unit tests
- [ ] tested upgrade from previous version

**Notes**:

### CODE REVIEW

File: pkg/engine/transform/transform.go [hunk 1]
```diff
@@ -384,6 +384,19 @@ func (t *Transformer) NormalizeResourcesForK8sMasterUpgrade(logger *logrus.Entry
 				if len(filteredResources) > 0 {
 					filteredResources = filteredResources[:len(filteredResources)-1]
 				}
+			} 
+		} else if resourceType == vnetResourceType {
+			if !ok {
+				logger.Warnf(fmt.Sprintf("Issue removing nsg dependency from vnet: %s", resourceName))
+			}
+			logger.Infoln(fmt.Sprintf("Removing all dependencies on vnet: %s", resourceName))
+			resourceMap[dependsOnFieldName] = []string{}
```
Comment (marosset) [inline pkg/engine/transform/transform.go]: I will clean this up to only remove the NSG dependency.

**Comment by Bhuvaneswari-Santharam** [comment]
lgtm

**Comment by CecileRobertMichon** [comment]
reverted because it broke upgrade :

```
Error: upgrading cluster: resources.DeploymentsClient#CreateOrUpdate: Failure sending request: StatusCode=400 -- Original Error: Code="InvalidTemplate" Message="Deployment template validation failed: 'The resource 'Microsoft.Network/networkSecurityGroups/k8s-master-40609989-nsg' is not defined in the template. Please see https://aka.ms/arm-template for usage details.'." AdditionalInfo=[{"info":{"lineNumber":0,"positionNumber":0,"snippet":""},"type":"TemplateViolation"}]
```

**Pull request merged**